src/ranged.cpp

```diff
@@ -258,25 +297,39 @@ bool player::handle_gun_damage( item &it, int shots_fired )
             }
         }
     }
-    if( curammo_effects.count( "BLACKPOWDER" ) ) {
-        if( !it.faults.count( fault_gun_blackpowder ) &&
-            it.faults_potential().count( fault_gun_blackpowder ) ) {
-            it.faults.insert( fault_gun_blackpowder );
-        }
-        if( one_in( firing->blackpowder_tolerance ) &&
-            it.faults_potential().count( fault_gun_clogged ) ) {
-            add_msg_player_or_npc( m_bad, _( "Your %s is clogged up with blackpowder fouling!" ),
-                                   _( "<npcname>'s %s is clogged up with blackpowder fouling!" ),
-                                   it.tname() );
-            it.faults.insert( fault_gun_clogged );
-            return false;
-        }
-        if( it.ammo_data()->ammo->recoil < firing->min_cycle_recoil && shots_fired > 0 ) {
-            add_msg_player_or_npc( m_bad, _( "Your %s fails to cycle!" ),
-                                   _( "<npcname>'s %s fails to cycle!" ),
-                                   it.tname() );
-            return false;
+    if( curammo_effects.count( "MUZZLE_SMOKE" ) || curammo_effects.count( "BLACKPOWDER" ) ) {
+        int blunderbussflintlock = 0;
+        int shot = 0;
+        if( it.dirt < 500 ) {
+            for( const ammotype &ammo : it.type->gun->ammo ) {
+                std::string compammo = ammo.c_str();
+                if( compammo == "shot" ) {
+                    shot = shot + 1;
+                } else if( compammo == "flintlock" || compammo == "blunderbuss" ) {
+                    blunderbussflintlock = blunderbussflintlock + 1;
+                }
+            }
+            if( blunderbussflintlock > 0 ) {
+                it.dirt = it.dirt + 3;
+            } else if( shot > 0 ) {
+                it.dirt = it.dirt + 10;
+            } else {
+                it.dirt = it.dirt + 22;
+            }
+            if( it.dirt > 500 ) {
+                it.dirt = 500;
+            }
+            if( !it.has_fault( fault_gun_blackpowder ) && !it.has_fault( fault_gun_clogged ) ) {
+                it.faults.insert( fault_gun_blackpowder );
+            }
         }
+        //      debugmsg( string_format( "%d", (it.dirt ) ) ); // gets the current dirt level of the gun.
```

Comment:
Please remove all commented-out code (or uncomment it if you need it working).

Reply:
ok i think i got all of them (but haven't pushed yet).